resources/services/lightsail/instance_snapshots.go

```diff
@@ -0,0 +1,301 @@
+					{
+						Name:        "state",
+						Description: "Describes the status of the disk",
+						Type:        schema.TypeString,
+					},
+					{
+						Name:        "support_code",
+						Description: "The support code",
+						Type:        schema.TypeString,
+					},
+					{
+						Name:        "tags",
+						Description: "The tag keys and optional values for the resource",
+						Type:        schema.TypeJSON,
+						Resolver:    resolveInstanceSnapshotFromAttachedDisksTags,
+					},
+				},
+				Relations: []*schema.Table{
+					{
+						Name:        "aws_lightsail_instance_snapshot_from_attached_disk_add_ons",
+						Description: "Describes an add-on that is enabled for an Amazon Lightsail resource",
+						Resolver:    fetchLightsailInstanceSnapshotFromAttachedDiskAddOns,
+						Columns: []schema.Column{
+							{
+								Name:        "instance_snapshot_from_attached_disk_cq_id",
+								Description: "Unique CloudQuery ID of aws_lightsail_instance_snapshot_from_attached_disks table (FK)",
+								Type:        schema.TypeUUID,
+								Resolver:    schema.ParentIdResolver,
+							},
+							{
+								Name:        "name",
+								Description: "The name of the add-on",
+								Type:        schema.TypeString,
+							},
+							{
+								Name:        "next_snapshot_time_of_day",
+								Description: "The next daily time an automatic snapshot will be created",
+								Type:        schema.TypeString,
+							},
+							{
+								Name:        "snapshot_time_of_day",
+								Description: "The daily time when an automatic snapshot is created",
+								Type:        schema.TypeString,
+							},
+							{
+								Name:        "status",
+								Description: "The status of the add-on",
+								Type:        schema.TypeString,
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+// ====================================================================================================================
+//                                               Table Resolver Functions
+// ====================================================================================================================
+
+func fetchLightsailInstanceSnapshots(ctx context.Context, meta schema.ClientMeta, parent *schema.Resource, res chan<- interface{}) error {
+	var input lightsail.GetInstanceSnapshotsInput
+	c := meta.(*client.Client)
+	svc := c.Services().Lightsail
+	for {
+		response, err := svc.GetInstanceSnapshots(ctx, &input, func(options *lightsail.Options) {
+			options.Region = c.Region
+		})
+		if err != nil {
+			return diag.WrapError(err)
+		}
+		res <- response.InstanceSnapshots
+		if aws.ToString(response.NextPageToken) == "" {
+			break
+		}
+		input.PageToken = response.NextPageToken
+	}
+	return nil
+}
+func resolveInstanceSnapshotsTags(ctx context.Context, meta schema.ClientMeta, resource *schema.Resource, c schema.Column) error {
+	r := resource.Item.(types.InstanceSnapshot)
+	tags := make(map[string]string)
+	client.TagsIntoMap(r.Tags, tags)
```

Comment:
should it be done this way? can't it be done via cq-gen?

Reply:
In this case we are not marshaling tag object directly. It would look next `[{"key":"test","value":"test"},{"key1":"test","value":"test"}]`  the resolver above makes it look next `{"key":"test", "key1":"test"}`.

Reply:
Maybe we can do a generic resolver instead here?